awswrangler/s3/_write_parquet.py

```diff
@@ -53,22 +53,29 @@ def _new_writer(
     use_threads: bool | int,
 ) -> Iterator[pyarrow.parquet.ParquetWriter]:
     writer: pyarrow.parquet.ParquetWriter | None = None
-    if not pyarrow_additional_kwargs:
+    if pyarrow_additional_kwargs is None:
         pyarrow_additional_kwargs = {}
-    if "coerce_timestamps" not in pyarrow_additional_kwargs:
-        pyarrow_additional_kwargs["coerce_timestamps"] = "ms"
-    if "flavor" not in pyarrow_additional_kwargs:
-        pyarrow_additional_kwargs["flavor"] = "spark"
-    if "version" not in pyarrow_additional_kwargs:
+    is_client_side_encryption_materials_present = (
+        "crypto_factory" in pyarrow_additional_kwargs
+        and "kms_connection_config" in pyarrow_additional_kwargs
+        and "encryption_config" in pyarrow_additional_kwargs
+    )
+    pyarrow_additional_settings = {
```

Comment:
This is effectively limiting the Arrow kwargs to those listed here.

For instance if a user wants to pass [data_page_version](https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetWriter.html) as a parameter, it will just be ignored.

It wouldn't be with the way it was setup before